### ISSUE TITLE
enumerate with VectorOfArray

### ISSUE THREAD

**Issue by moesphere** [SciML/RecursiveArrayTools.jl#175]
Enumerating the following MWE gives an unexpected result:

```julia
[(i,x) for (i,x) in enumerate(VectorOfArray([[1,2,3]]))]
````
![grafik](https://user-images.githubusercontent.com/19595387/145692782-c396cf0a-9ef5-4eb9-a8e0-5e8a5cf568cd.png)

**Comment by ChrisRackauckas** [comment]
Yeah... iteration is weird. Iteration iterates over the vectors, not the elements of the vector. This was a decision made a long long time ago which was probably wrong, but would be very breaking so it would be necessary to tie it to DiffEq v7.
